bpf/lib/generic.h

```diff
@@ -37,6 +37,8 @@ struct msg_generic_kprobe {
 	__u64 id;
 	__u64 thread_id;
 	__u64 action;
+	__u32 url_id; // only one URL action can be fired per match
+	__u32 fqdn_id; // only one DNS action can be fired per match
```

Comment:
From the bpf-side, there is no difference between the URL and FQDN. It's just something that needs to happen in userspace. I think it would be better if we just use a single id for all the user-space actions and let the agent figure things out.

This will also make it easier to add more use-space actions (e.g., freeze).

I think this would simplify  the user-space side as well: having a single id table that has all the registered actions.

Reply:
True. I can do that.